tensorflow/python/training/optimizer.py

```diff
@@ -292,6 +298,13 @@ def apply_gradients(self, grads_and_vars, global_step=None, name=None):
             "Variable must be a tf.Variable: %s" % v)
       if g is not None:
         self._assert_valid_dtypes([g, v])
+      # If all is well, append g and v to corresponding list separately 
+      grads_to_tensors_list.append(g)
+      vars_list.append(v) 
```

Comment:
Instead of maintaining two lists in sync, change this to maintain a single list of tuples, called e.g. `converted_grads_and_vars` (then use `converted_grads_and_vars` in place of `grads_and_vars` in the remainder of this function).